### ISSUE TITLE
userdata socket not receiving updates

### ISSUE THREAD

**Issue by jcageman** [sammchardy/python-binance#167]
Hi,

i am experiencing some cases in which i don't receive updates of trades via the userdata socket. I already applied the patch for ensuring the socket is kept alive. I also don't think the issue below is caused by it, because i do receive other updates in the mean time.

One legitimate case i found is a direct fill order in the reply of the limit order. Typically where it goes wrong now is that my api thinks it should cancel the order since its been overbid too many times and when it tries to do that the order is "unknown". I already created a fallback in this case and basically reset my script for these cases to ensure it remains correct. There still remain cases where i for example never think about cancelling my order (price goes up for example) and my program keeps thinking the order is pending, while it has already been processed. It seems to me this is a binance issue with the userdata socket. Any others experiencing this? How to resolve this correctly?

Thanks!

**Comment by sammchardy** [comment]
Hi @jcageman which patch did you apply?

Are you indicating that there are particular order types that reproduce this issue?

**Comment by jcageman** [comment]
I applied #134. I doubt it has to do with the order type, but i am mainly using limit orders in my api calls. Yesterday i noticed even after applying #134 the userdata socket stops updating without any error in the socket messages before it. I will keep monitoring and hopefully i can find a cause.

**Comment by coire1** [comment]
Hi @jcageman and @sammchardy, after some days using the #134 I encountered the `listenKey` change that you handle in the master version.
I think that I identified the problem on master version. 
Specifically, when the library detect a [key change](https://github.com/sammchardy/python-binance/blob/4a433f83aba1717e9b0a0c9439125bada91b5a0c/binance/websockets.py#L462), it tries to reconnect to socket with another key, and not the new one already received.
I'm preparing a PR to fix this issue. On which branch I have to base the PR?
